machine_learning_hep/processer.py

```diff
@@ -478,7 +478,7 @@ def skim(self, file_index):
                 dfgensk = dfquery(dfgensk, self.s_gen_skim[ipt])
                 write_df(dfgensk, self.mptfiles_gensk[ipt][file_index])
 
-            if dfgen_sl is not None:
```

Comment:
What was wrong with the check for not None?

Reply:
dfgen_sl is not defined